packages/drivers/driver-base/src/documentDeltaConnection.ts

```diff
@@ -504,7 +504,8 @@ export class DocumentDeltaConnection
 					const normalizedError = this.addPropsToError(failError);
 					this.logger.sendErrorEvent({ eventName: "CloseSocketError" }, normalizedError);
 				}
-				reject(err);
+				const errorToReject = err instanceof Error ? err : new Error(JSON.stringify(err));
+				reject(errorToReject);
```

Comment:
This seems like a potential behavior change, I'd revert this and all the similar refactors.

Reply:
Reverted most of them. There were a couple that seemed safe in some test code.